crates/ra_hir_ty/src/display.rs

```diff
@@ -269,9 +402,12 @@ impl HirDisplay for ApplicationTy {
 }
 
 impl HirDisplay for ProjectionTy {
-    fn hir_fmt(&self, f: &mut HirFormatter) -> fmt::Result {
+    fn hir_fmt<R>(&self, f: &mut HirFormatter<R>) -> Result<(), R::Error>
+    where
+        R: RenderTarget,
+    {
         if f.should_truncate() {
-            return write!(f, "{}", TYPE_HINT_TRUNCATION);
+            return Ok(write!(f, "{}", TYPE_HINT_TRUNCATION)?);
```

Comment:
I think you could make `HirFormatter::write_fmt` return the correct error type directly to avoid a lot of error type conversions. (Note that we're making use of `write!`s duck-typing here, `f` is the `HirFormatter` and `write!` is just calling the `write_fmt` method defined in this file.)

Reply:
Hm, yeah, true. Very good idea :)